rpi/logger.py

```diff
@@ -0,0 +1,109 @@
+import datetime
+from datetime import (datetime, timedelta)
+import os
+from git import Repo
+
+
+log_folder = "/home/pi/doorbotlog"
+logRepoURL = "https://github.com/site-3/doorbotlog"
+
+class Logger:
+    last_pushed:datetime
+    verbose_level:int = 5#0-5. 5 being for debugging, 0 being for errors that always need to be logged
+    def __init__(self, verbose:int=5, debugMode:bool=False, setupRepo:bool=True):
+        #this should probably throw an error if the logging is outside of the expected range. But IDC
+        self.verbose_level = verbose
+        if not setupRepo:
+            return
+        self.repo = Repo(log_folder)#get the local repo files connected
+        self.origin = self.repo.remote() #get our remote connection setup
+
+        #check that the remote connection is up, and if it isn't, log that!
+        if not self.origin.exists():
+            self.log("origin could not be connected to!", verbose=0)
+        self.origin.fetch()#this can give data apparently, but for now, we just update our stuff with it
+        # Create local branch "master" from remote "master".
+        # Set local "master" to track remote "master.
+        # Check out local "master" to working tree.
+        if debugMode:
+            self.repo.create_head("testing", self.origin.refs.testing)
+            trackedBranch = self.repo.heads.testing.tracking_branch()
+            if trackedBranch == None or trackedBranch.name != "origin/testing":
+                #this will cause bugs if we call it when it's already the tracking branch
+                self.repo.heads.testing.set_tracking_branch(self.origin.refs.testing)
+            self.repo.heads.testing.checkout()
+        else:
+            self.repo.create_head("master", self.origin.refs.master)
+            trackedBranch = self.repo.heads.master.tracking_branch()
+            if trackedBranch == None or trackedBranch.name != "origin/master":
+                self.repo.heads.master.set_tracking_branch(self.origin.refs.master)
+            self.repo.heads.master.checkout()
+        self.origin.pull()#make sure our copies are up to date and will play nice
```

Comment:
Will this fail if there is a merge conflict?

Reply:
quite possibly. Currently, it fails if there's a merge, even without conflicts. There also shouldn't be merge conflicts, because at no point should anyone be changing the logs. Any other changes to the log repo shouldn't cause conflicts.